src/NexusMods.App.UI/WorkspaceSystem/Workspace/WorkspaceViewModel.cs

```diff
@@ -408,7 +411,22 @@ private void OpenPageReplaceTab(PageData pageData, OpenPageBehavior.ReplaceTab r
         if (!tab.Contents.ViewModel.CanClose()) return;
 
         // Replace the tab contents
-        var newTabPage = _factoryController.Create(pageData, WindowId, Id, panel.Id, tab.Id);
+
+        Page newTabPage;
+
+        try
+        {
+            newTabPage = _factoryController.Create(pageData, WindowId, Id, panel.Id, tab.Id);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Exception creating page for factory `{FactoryId}` with context `{Context}`", pageData.FactoryId, pageData.Context);
+
+            var defaultPageData = GetDefaultPageData();
+            OpenPageReplaceTab(defaultPageData, replaceTab, selectTab, checkOtherPanels: false);
```

Comment:
I think it would be better not to recall the function inside the catch, and just use the data newTabPage data instead.
Unless there is a reason you didn't do that?

Reply:
We can do that.